doc/api/vm.markdown

```diff
@@ -301,6 +301,34 @@ e.g. `(0,eval)('code')`. However, it also has the following additional options:
 - `timeout`: a number of milliseconds to execute `code` before terminating
   execution. If execution is terminated, an [`Error`][] will be thrown.
 
+## Example: Run a Server within a VM
+
+Since the context of `.runInThisContext()` actually means the V8 context, you
+will need to apply additional objects to it. Say, you want to run a simple
+web server based on the `http` module, you can achieve this by doing something
+like this:
+
+```js
+'use strict'
+const vm = require('vm')
+
+let code = `
+(function(require) {
```

Comment:
minor nit... having the `let code =` on a separate line with no indenting on the next line makes it a bit confusing.